crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -127,23 +151,47 @@ fn match_prepared_and_settled_orders(
 fn match_prepared_and_settled_amms(
     prepared_amms: Vec<Liquidity>,
     settled_amms: HashMap<usize, UpdatedAmmModel>,
-) -> Result<Vec<ExecutedAmm>> {
+) -> Result<Vec<(Option<ExecutionPlanCoordinatesModel>, Execution)>> {
     settled_amms
         .into_iter()
         .filter(|(_, settled)| settled.is_non_trivial())
         .flat_map(|(index, settled)| settled.execution.into_iter().map(move |exec| (index, exec)))
-        .sorted_by_key(|(_, a)| a.exec_plan.clone())
         .map(|(index, settled)| {
-            Ok(ExecutedAmm {
-                order: prepared_amms
-                    .get(index)
-                    .ok_or_else(|| anyhow!("Invalid AMM {}", index))?
-                    .clone(),
-                input: (settled.buy_token, settled.exec_buy_amount),
-                output: (settled.sell_token, settled.exec_sell_amount),
-            })
+            Ok((
+                settled.exec_plan.clone(),
+                Execution::ExecutionAmm(ExecutedAmm {
+                    order: prepared_amms
+                        .get(index)
+                        .ok_or_else(|| anyhow!("Invalid AMM {}", index))?
+                        .clone(),
+                    input: (settled.buy_token, settled.exec_buy_amount),
+                    output: (settled.sell_token, settled.exec_sell_amount),
+                }),
+            ))
         })
-        .collect()
+        .collect::<Result<Vec<(Option<ExecutionPlanCoordinatesModel>, Execution)>>>()
+}
+
+fn merge_and_order_executions(
```

Comment:
Would be nice to explicitly say (in a comment) what happens with executions that don't have coordinates (do they go first or last).